### ISSUE TITLE
Certificate.inspect busted

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2127]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2127|
|Type|defect|
|Created|2006-09-25 19:49:59Z|

Watch:

```
>>> from twisted.internet.ssl import Certificate
>>> Certificate.loadPEM(file('Projects/Twisted/trunk/doc/core/examples/server.pem').read()).inspect()
Traceback (most recent call last):
  File "<stdin>", line 1, in ?
  File "/home/exarkun/Projects/Twisted/trunk/twisted/internet/_sslverify.py", line 491, in inspect
    return '\n'.join(self._inspect(), self.getPublicKey().inspect())
TypeError: join() takes exactly one argument (2 given)
>>> 
```

<details><summary>Searchable metadata</summary>

```
trac-id__2127 2127
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__Core_2_5 Core-2.5
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1159213799000000 1159213799000000
changetime__1159962690000000 1159962690000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Fixed and ready for review in certificate-inspect-2127

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml set owner to @exarkun|
|-|-|

Looks good. Please add a docstring for inspect(), then merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [18386]) Merge certificate-inspect-2127

Author: exarkun
Reviewer: jml
Fixes [#2127](https://github.com/twisted/twisted/issues/2127)

Add test coverage for Certificate.inspect and fix it to not raise an exception.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
